crates/interledger-settlement-engines/src/engines/ethereum_ledger/eth_engine.rs

```diff
@@ -0,0 +1,943 @@
+                    // connector about all the transactions in that block that are sent to our account
+                    let notify_eth_txs_fut = checked_blocks
+                        .map(move |block_num| self_clone.notify_eth_txs_in_block(block_num));
+
+                    // combine all the futures for that range of blocks
+                    Either::B(join_all(notify_eth_txs_fut))
+                };
+
+                notify_all_txs_fut.and_then(move |ret| {
+                    debug!("Transactions settled {:?}", ret);
+                    // now that all transactions have been processed successfully, we
+                    // can save `fetch_until` as the latest observed block
+                    store_clone.save_recently_observed_block(fetch_until)
+                })
+            })
+    }
+
+    /// Submits an ERC20 transfer object's data to the connector
+    // todo: Try combining the body of this function with `notify_eth_transfer`
+    fn notify_erc20_transfer(
+        &self,
+        transfer: ERC20Transfer,
+        token_address: Address,
+    ) -> impl Future<Item = (), Error = ()> {
+        let store = self.store.clone();
+        let tx_hash = transfer.tx_hash;
+        let self_clone = self.clone();
+        let addr = Addresses {
+            own_address: transfer.from,
+            token_address: Some(token_address),
+        };
+        let amount = transfer.amount;
+        store
+            .check_if_tx_processed(tx_hash)
+            .map_err(move |_| error!("Error when querying store about transaction: {:?}", tx_hash))
+            .and_then(move |processed| {
+                if !processed {
+                    Either::A(
+                        store
+                            .load_account_id_from_address(addr)
+                            .and_then(move |id| {
+                                self_clone.notify_connector(
+                                    id.to_string(),
+                                    amount.low_u64(),
+                                    tx_hash,
+                                )
+                            })
+                            .and_then(move |_| {
+                                // only save the transaction hash if the connector
+                                // was successfully notified
+                                store.mark_tx_processed(tx_hash)
+                            }),
+                    )
+                } else {
+                    Either::B(ok(())) // return an empty future otherwise since we want to skip this transaction
+                }
+            })
+    }
+
+    fn notify_eth_txs_in_block(&self, block_number: u64) -> impl Future<Item = (), Error = ()> {
+        debug!("Getting txs for block {}", block_number);
+        let self_clone = self.clone();
+        // Get the block at `block_number`
+        self.web3
+            .eth()
+            .block(BlockNumber::Number(block_number as u64).into())
```

Comment:
```suggestion
            .block(BlockNumber::Number(block_number).into())
```
Don't think that cast does anything